### ISSUE TITLE
Address Nessus baselinue compliance findings by 2022-06-20 CG379

### ISSUE THREAD

**Issue by pburkholder** [cloud-gov/cg-harden-boshrelease#54]
In order to submit accurate ConMon reports, we need to dig into the findings from version 2.x of the Nessus CIS benchmark findings. 

This is essential a repeat of our work for https://github.com/cloud-gov/product/issues/1609. That ticket read:

In order to meet FedRAMP compliance objectives, we want to close all findings from Nessus CIS benchmark compliance scan, or show that they are operational requirements or false positives.

At this point there are about 64 failed checks. When we need to we can use our parse-nessus script to extract the full description of all the fixes.

We will also need to revisit the existing hardening steps and remove obsolete ones.

As we do this work, check in with compliance to update our POA&Ms and Deviation Requests to remove obsolete entries and put new POA&Ms and Deviation Requests on track to the JAB. We may want to iterate through on subsystems until we're complete.

Keywords: CG379 CG-379 

## Acceptance Criteria

- [ ] A nessus output that is as clean a practicable
- [ ] A cg-harden-boshrelease that is cleared of obsolete entries
- [ ] An updated POA&M Tracker with new entries added and closed ones moved and documented
- [ ] An updated Deviation Request tracker with False Positive and OR entries 

Also, we need to review the SSP and be sure 
- [ ] that we are correctly specifying our use of Bionic Level One or Level Two compliance benchmarchs when we refer to the CIS Benchmarks. We may need to correct the SSP, as we only need L1 for a Moderate Impact system. 
- [ ] if we change from using chrony instead of ntp, for example, that we reference the correct service.

## Implementation steps

- [ ] We need a tracker spreadsheet [like this one from Oct 2021](https://docs.google.com/spreadsheets/d/1AcEgCKLZ-9p0E_qyFk3m47J80uwNBgpBByXDEMG5x3E/edit#gid=0) - We can start with CSV from scan and import into Google Sheets.
- [ ] Then do the work...

--- 

## Security considerations
Yes. We need to be careful about sharing any findings that might be relevant to malicious actors.

**Comment by pburkholder** [comment]
I've created a [spreadsheet with the compliance findings](https://docs.google.com/spreadsheets/d/1-nSDeThi-Bk5Yh0O12EGRd4llogBuR6bZX4jipFmdMI/edit#gid=1209745293) and I've uploaded a copy of the [Ubuntu 18.04 CIS benchmarks](https://drive.google.com/file/d/1Wh3d2hAFuYvrDfaC7x32QMRchl5ar6pC/view?usp=sharing).

At this point it would help to have an engineer pick this up. This is a good first issue for someone learning BOSH and familiar with Linux/Ubuntu, and also needing to learn more about how our systems work. What we need is for each finding to be evaluated as one of:

* a false positive, because the logic of the plugin is faulty
   * example: The WARNING for not having package repositories set up
* a false positive, because we meet the requirement/intent of the CIS control
  * example: "4.2.1.2 Ensure rsyslog Service is enabled" : [FAILED] - We meet the log shipping requirement by other means...
* a fix that we can make to clear the finding.
  * example: "4.2.1.3 Ensure logging is configured - 'mail.err /var/log/mail.err'" : [FAILED]. This is also a false positive, because we don't have smtp running on (most) machines, but it's easier to make findings go away with code than it is to document why they're false positives.
  * For these, we'll need a PR (one big PR with most of the fixes will be fine) to the https://github.com/cloud-gov/cg-harden-boshrelease repo
* not sure: We'll need to meet to discuss edge cases, and probably pull in a more experience engineer.

This is a 4-16 hour task, and I'd be happy to do it, but I'm on the critical path for other work, so it would be good to hand off.

**Comment by bengerman13** [comment]
Keep in mind while working on these that the `not sure` and `a fix that we can make to clear the finding` could spawn into new issues so we can keep progress moving - e.g. if we find a thing we know needs a fix, but that fix is complicated, make a new issue and consider it done for the purpose of this issue.

**Comment by pburkholder** [comment]
@dandersonsw and I are working on this.

**Comment by mogul** [comment]
@dandersonsw is blocked by missing Mac access.

**Comment by mogul** [comment]
@dandersonsw says the majority of this is done; needs to sync with Peter to review results and the outstanding PR.

**Comment by dandersonsw** [comment]
@pburkholder I've finished addressing the various findings. I'll pass this back to you for review.

**Comment by pburkholder** [comment]
Closing this and moving remaining work to https://github.com/cloud-gov/private/issues/718
